Srtrace/react phone catalog solution

SrTrace opened the pull request:
[LINK](https://srtrace.github.io/react_phone-catalog/)

SanyaBratashchuk reviewed (changes requested):
data fetching is broken 
![image](https://github.com/user-attachments/assets/4bb0ad77-261d-40fc-9a66-f2f7a1246da6)

SanyaBratashchuk reviewed (changes requested):
![image](https://github.com/user-attachments/assets/bc904b22-2600-4f35-b4d0-3e5aef365916)
color buttons should have different colors

SrTrace commented:
I have changed the button colors according to Apple's color palette. Now it looks a little bit better :)

volodymyr-soltys97 reviewed (changes requested):
Good job 👍 
Let's improve your page
1. Add transition for hover effects here 
<img width="212" alt="image" src="https://github.com/user-attachments/assets/c17ff0ca-aa13-431a-9d48-6c258c56b4dd">

2. Don't need to hide menu when search is open 
<img width="1357" alt="image" src="https://github.com/user-attachments/assets/fb851b75-06d7-4656-b84d-4786080aa20f">

3. These buttons must be are on the same vertical line 
<img width="284" alt="image" src="https://github.com/user-attachments/assets/b066145d-1a95-45b5-b33e-b14a92410f17">

SrTrace commented:
Okay, it’s fixed! I added a hover transition to the card button, made the navigation menu visible when the search is open, and fixed the alignment for CartItem.

volodymyr-soltys97 reviewed (changes requested):
Almost done!
To improve:
It is necessary to remove the limitation of the maximum quantity of 10 units, the user must choose the quantity he needs, even if it is more than 10
<img width="808" alt="image" src="https://github.com/user-attachments/assets/c25df789-1db6-43e6-acdd-75e0726e6e2b">

SrTrace commented:
The limit on the quantity of items in the cart has been fixed.

volodymyr-soltys97 reviewed (approved):
Great! 🔥
